### ISSUE TITLE
[cloud_firestore] Error pop-up when app restart after changing cloud_firestore version (v0.13.4 + 2 --> v0.13.5)

### ISSUE THREAD

**Issue by exptk3** [FirebaseExtended/flutterfire#2434]
*** Error pop-up when app restart after changing cloud_firestore version
- Before change: 0.13.4 + 2
- After change: 0.13.5
- The app does not start with an error message pop-up on restart after deactivating the app with the back button without logging out.
- Error message : <App name> keeps stopping ('! App info' button, 'x Close app' button)
- This bug disappeared as the package was restored to version 0.13.4 + 2.

<Environment>
Flutter 1.12.13+hotfix.9 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f139b11009 (4 weeks ago) • 2020-03-30 13:57:30 -0700
Engine • revision af51afceb8
Tools • Dart 2.7.2

**Comment by Ehesp** [comment]
Without any logs about what the error is, it's hard to help. What platform are you on?

**Comment by exptk3** [comment]
> Without any logs about what the error is, it's hard to help. What platform are you on?

My laptop OS is Mac.
This is last log. After this log, app crash, emulator disconnected and can not see any more logs.

D/AndroidRuntime( 9845): Shutting down VM
E/AndroidRuntime( 9845): FATAL EXCEPTION: main
E/AndroidRuntime( 9845): Process: com.xxx.xxx, PID: 9845
E/AndroidRuntime( 9845): java.lang.NullPointerException: Attempt to invoke virtual method 'void io.flutter.plugin.common.MethodChannel.invokeMethod(java.lang.String, java.lang.Object)' on a null object reference
E/AndroidRuntime( 9845): 	at io.flutter.plugins.firebase.cloudfirestore.CloudFirestorePlugin$DocumentObserver.onEvent(CloudFirestorePlugin.java:429)
E/AndroidRuntime( 9845): 	at io.flutter.plugins.firebase.cloudfirestore.CloudFirestorePlugin$DocumentObserver.onEvent(CloudFirestorePlugin.java:402)
E/AndroidRuntime( 9845): 	at com.google.firebase.firestore.DocumentReference.lambda$addSnapshotListenerInternal$2(com.google.firebase:firebase-firestore@@21.3.0:504)
E/AndroidRuntime( 9845): 	at com.google.firebase.firestore.DocumentReference$$Lambda$3.onEvent(Unknown Source:6)
E/AndroidRuntime( 9845): 	at com.google.firebase.firestore.core.AsyncEventListener.lambda$onEvent$0(com.google.firebase:firebase-firestore@@21.3.0:42)
E/AndroidRuntime( 9845): 	at com.google.firebase.firestore.core.AsyncEventListener$$Lambda$1.run(Unknown Source:6)
E/AndroidRuntime( 9845): 	at android.os.Handler.handleCallback(Handler.java:883)
E/AndroidRuntime( 9845): 	at android.os.Handler.dispatchMessage(Handler.java:100)
E/AndroidRuntime( 9845): 	at android.os.Looper.loop(Looper.java:214)
E/AndroidRuntime( 9845): 	at android.app.ActivityThread.main(ActivityThread.java:7356)
E/AndroidRuntime( 9845): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 9845): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
E/AndroidRuntime( 9845): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
I/Process ( 9845): Sending signal. PID: 9845 SIG: 9

**Comment by Ehesp** [comment]
Thanks, could you try to isolate the code you're using which triggers this crash?

**Comment by exptk3** [comment]
> Thanks, could you try to isolate the code you're using which triggers this crash?

I found a step that triggers an app crash by taking a log between every detail of the app loading process. When loading the app, the data is stored in Firebase. At this stage, the vm is down.

* log
I/flutter (31797): setData(path & data.length) : xxxxxx
D/AndroidRuntime(31797): Shutting down VM
E/AndroidRuntime(31797): FATAL EXCEPTION: main

* This is error-causing setData code
await _service.setData(
        path: APIPath.xxx,
        data: xxxObject.toMap(),
      );

* Test
 - I tried to print the error of setData using try-catch statement, but the vm was down without error log.
 - Since this setData code was omitted, the app started normally when reloading after ending the back button.
 - It was confirmed whether the data (xxxObject) injected into setData was abnormal ?
 --> but the data was normal.
 - After 5 seconds of delay using await Future.delayed --> the app crashed after 5 second.
 - The setData process when loading the app worked perfectly fine in the previous version of cloud_firestore(v0.13.4+2).

I hope this helps you fix the bug. Thank you.
